feat: add AVS guide (#96)
hyperlane-xyz/v3-docs: docs/guides/avs-operator-guide.mdx

Next, to register your operator to Hyperlane AVS, you can register your operator key (generated via EigenLayer CLI) by creating an ECDSA signature and submitting it via the Hyperlane CLI:

```bash
 hyperlane avs register --operatorKeyPath <PATH_TO_OPERATOR_KEY> --chain holesky --key <HYP_KEY>
```

### Running the validator

Follow the validator guide [here](../operate/validators/run-validators.mdx) with the AWS setup to run the validator.

You can monitor Eigenlayer specific endpoints via the `--metrics` port you specified in the validator configuration. Read more about the example endpoints and what should be expected [here](https://docs.eigenlayer.xyz/eigenlayer/avs-guides/spec/api/).

The supported endpoints are:

| Endpoint                                  | Description                                |
| ----------------------------------------- | ------------------------------------------ |
| **GET /eigen/node**                       | for node info                              |
| **GET /eigen/node/health**                | for node health                            |
| **GET /eigen/node/services**              | for a list of services the node is running |

> | **GET /eigen/node/`{service_id}`/health** | for health of a specific service           |

Reviewer: What is `service_id` here? IIRC, we didn't really have subservices, so maybe lets just omit this information?

Author: We have serviceId `hyperlane-validator-indexer` and `hyperlane-validator-submitter` hardcoded. So, I'll remove it until it's not hardcoded.

    function handleChallenge(address operator) external;
}
```

This configuration should be immutable and accessible for any AVS operator to view and opt into. `handleChallneger` implicitly encodes how you expect the challenge from the source chain (say Arbitrum) is delivered to Ethereum Mainnet where the AVS contracts and the IRemoteChallenger live. For Arbitrum, it can be the rollup's native bridge and for another L1, it can be a committee-based solution. Our interface is challenger method agnostic to allow for flexibility in this design area. A reasonable `challengeDelayBlocks` would be slightly longer than in a week in the case of using the native bridge.

As an operator, you are expected to review the different `IRemoteChallenger` contracts and assess their risk and rewards. If interested, you can enroll into one or multiple challengers directly from the `HyperlaneServiceManager` contract or using our CLI:

```bash
cast send --private-key <PRIVATE_KEY> $SERVICE_MANAGER "enrollIntoChallenger(address)" <CHALLENGER_ADDRESS> --rpc-url <RPC_URL>
```

NOTE: The IRemoteChallenger implementations are not yet live in production.

## Register your operator

:::warning

> There is a distinction between the operator key and the AVS signing key. The operator key is meant for just interacting with the AVS contracts whereas the AVS signing key is used for active operations which in our case will be signing checkpoints with a Hypelane validator. As on May 15th, 2024. Eigenlayer doesn't support attesting to an AVS signing key while registering as an operator but that will be changing in short order. More details about the keys [here](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/key-management/intro#operator-keys).

Reviewer 2: ```suggestion
There is a distinction between the operator key and the AVS signing key. The operator key is meant for just interacting with the AVS contracts whereas the AVS signing key is used for active operations, which in our case will be signing checkpoints with a Hyperlane Validator. As of May 15th 2024, EigenLayer doesn't support attesting to an AVS signing key while registering as an operator but that will be changing in short order. You can read more details about the keys [here](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/key-management/intro#operator-keys).
```

:::

> For registering your operator to Hyperlane AVS, you can register your operator key generated using Eignelayer CLI by creating an EIP-191 compliant signature and submitting it via our CLI:

Reviewer 2: ```suggestion
To register your operator to Hyperlane AVS, you can register your operator key (generated via EigenLayer CLI) by creating an EIP-191 compliant signature and submitting it via the Hyperlane CLI:
```

```bash
 hyperlane avs register --operatorKeyPath <PATH_TO_OPERATOR_KEY> --chain holesky --key <PRIVATE_KEY_FOR_SENDING_TX>
```

> Note: PRIVATE_KEY_FOR_SENDING_TX can be any third-party key and doesn't need to be operator key.

Reviewer 2: admonition again
